### ISSUE TITLE
Inconsistent Leak Sensor behavior

### ISSUE THREAD

**Issue by danielduerr** [pyinsteon/pyinsteon#210]
Hi,

I have 5 Insteon Leak Sensors in my installation.  When I (re)start Home Assistant, they all report Dry conditions.  Over ~48 hours, all but one of them will randomly report Moist conditions and will remain in that state until the next restart.  I have manually verified that there are no actual leaks/moisture conditions.  And, it is always the same device (Water Heater) that remains in a correct Dry state, versus the other 4 devices which always end up in the incorrect Moist state.

All 5 of these devices were previously setup/ALDB linked to the PLM before I migrated my PLM and Insteon network to Home Assistant.  They are all the exact same model and FW version, and had identical ALDB link records.  They had been installed in these same locations for several years and had worked as expected in my previous PHP-based home automation controller until migrating them in to Home Assistant.

My suspicion is that the issue here is one of the following:

1. The Home Assistant (pyinsteon) driver for this device type is not working as expected; or
2. The Home Assistant (pyinsteon) driver for this device type doesn't work as expected with my ALDB links; or
3. The initial adoption of these devices into Home Assistant (pyinsteon) modified their ALDB links from a working state into a non-working state

Here is an example of the ALDB links for the working (Water Heater) device:
![Screen Shot 2022-06-23 at 8 15 13 AM](https://user-images.githubusercontent.com/4553104/175297133-9819cb57-be3a-4b7e-a64e-9c0dabf939a0.png)

Here is an example of the ALDB links for one of the other erratic devices:
![Screen Shot 2022-06-23 at 8 15 27 AM](https://user-images.githubusercontent.com/4553104/175297227-fc627ba8-b7e1-467e-9e02-322f9b3a4123.png)

Here is the json definition of the working (Water Heater) device:
```
  {
    "address": "3fc195",
    "cat": 16,
    "subcat": 8,
    "firmware": 0,
    "engine_version": 2,
    "aldb_status": 1,
    "aldb": {
      "4095": {
        "memory": 4095,
        "in_use": true,
        "controller": false,
        "high_water_mark": false,
        "bit5": true,
        "bit4": false,
        "group": 0,
        "target": "191cd6",
        "data1": 0,
        "data2": 0,
        "data3": 0
      },
      "4087": {
        "memory": 4087,
        "in_use": true,
        "controller": true,
        "high_water_mark": false,
        "bit5": true,
        "bit4": false,
        "group": 1,
        "target": "191cd6",
        "data1": 254,
        "data2": 0,
        "data3": 1
      },
      "4079": {
        "memory": 4079,
        "in_use": true,
        "controller": true,
        "high_water_mark": false,
        "bit5": true,
        "bit4": false,
        "group": 2,
        "target": "191cd6",
        "data1": 254,
        "data2": 0,
        "data3": 2
      },
      "4071": {
        "memory": 4071,
        "in_use": true,
        "controller": true,
        "high_water_mark": false,
        "bit5": true,
        "bit4": false,
        "group": 4,
        "target": "191cd6",
        "data1": 254,
        "data2": 0,
        "data3": 4
      },
      "4063": {
        "memory": 4063,
        "in_use": false,
        "controller": false,
        "high_water_mark": true,
        "bit5": false,
        "bit4": false,
        "group": 0,
        "target": "000000",
        "data1": 0,
        "data2": 0,
        "data3": 0
      }
    },
    "operating_flags": {},
    "properties": {
      "led_brightness": 100,
      "cleanup_report_on": true,
      "ignore_jumper_on": true,
      "two_groups_on": true,
      "repeat_open_on": true,
      "repeat_closed_on": false,
      "led_on": false,
      "link_to_ff_group": true,
      "program_lock_on": false
    },
    "first_mem_addr": 4095
  },
```

And here is the json definition of the other erratic device:
```
  {
    "address": "3f59a0",
    "cat": 16,
    "subcat": 8,
    "firmware": 0,
    "engine_version": 2,
    "aldb_status": 1,
    "aldb": {
      "4095": {
        "memory": 4095,
        "in_use": true,
        "controller": false,
        "high_water_mark": false,
        "bit5": true,
        "bit4": false,
        "group": 0,
        "target": "191cd6",
        "data1": 0,
        "data2": 0,
        "data3": 0
      },
      "4087": {
        "memory": 4087,
        "in_use": true,
        "controller": true,
        "high_water_mark": false,
        "bit5": true,
        "bit4": false,
        "group": 1,
        "target": "191cd6",
        "data1": 254,
        "data2": 0,
        "data3": 1
      },
      "4079": {
        "memory": 4079,
        "in_use": true,
        "controller": true,
        "high_water_mark": false,
        "bit5": true,
        "bit4": false,
        "group": 2,
        "target": "191cd6",
        "data1": 254,
        "data2": 0,
        "data3": 2
      },
      "4071": {
        "memory": 4071,
        "in_use": true,
        "controller": true,
        "high_water_mark": false,
        "bit5": true,
        "bit4": false,
        "group": 4,
        "target": "191cd6",
        "data1": 254,
        "data2": 0,
        "data3": 4
      },
      "4063": {
        "memory": 4063,
        "in_use": false,
        "controller": false,
        "high_water_mark": true,
        "bit5": false,
        "bit4": false,
        "group": 0,
        "target": "000000",
        "data1": 0,
        "data2": 0,
        "data3": 0
      }
    },
```

Does anything jump out here?

Thanks in advance for your assistance!

**Comment by teharris1** [comment]
Once Home Assistant 2022.7.4 is released, please upgrade to that release version. We can then debug any of your leak sensors and see what is causing this issue. To place the device in debug mode enter the following into your `configuration.yaml` file:
```
logger:
  default: warn
  logs:
    pyinsteon.<device id>: debug
    pyinsteon.<another device id>: debug
```
Where the device id is the Insteon device such as 3fc195. Use all lower case and no decimal points in the device id.  Restart HA and let it run for 24 - 48 hours then send me the output of the logs.

**Comment by danielduerr** [comment]
@teharris1 I have applied the 2022.7.4 release, as well as the logging lines in `configuration.yaml` you requested.  Everything has been restarted with this in place, and all sensors are legitimately reporting a Dry state.  In 24-48 hours I can get you the logs.  Can you tell me how to fetch them for you?  Haven't used HA's log interface much.

**Comment by danielduerr** [comment]
@teharris1 attached, please find 2 log files per your last request. Each log file has been filtered to just the pyinsteon events.  Each log file represents a full HA startup where the leak sensors were all in a Dry state, and all but 1 ultimately went into the Wet state as I reported above.

[leaksensor.home-assistant.log.1.zip](https://github.com/pyinsteon/pyinsteon/files/9159967/leaksensor.home-assistant.log.1.zip)
[leaksensor.home-assistant.log.zip](https://github.com/pyinsteon/pyinsteon/files/9159966/leaksensor.home-assistant.log.zip)

**Comment by danielduerr** [comment]
@teharris1 thanks, I'll take that as good news. I admittedly didn't understand the part about having to tap the set button once to get it back to a Dry state. But I read the Quick Start guide on it and sure enough it describes that process. I restarted HA and am waiting for them to start heartbeating in as Wet, at which point I'll try what you suggested and report back on how it works.

**Comment by danielduerr** [comment]
@teharris1 well, it looks like you were spot-on.  I pressed the Set button manually on all but one of the problematic leak sensors and sure enough, they have all remained in a Dry state.  The one last problematic one (hidden in a bath tub motor compartment) went off as expected last night.  This morning, I opened up the compartment and pressed the Set button on it manually.  Voila, it went back to a Dry state and I even got my first push notification from my automation that tells me when they go back to dry.
